qiskit/test/mock/utils/backend_converter.py

```diff
@@ -0,0 +1,142 @@
+        # Parse instructions
+        gates = {}
+        for gate in props_dict["gates"]:
+            name = gate["gate"]
+            if name in name_mapping:
+                if name not in gates:
+                    gates[name] = {}
+            elif name not in custom_gates:
+                custom_gate = Gate(name, len(gate["qubits"]), [])
+                custom_gates[name] = custom_gate
+                gates[name] = {}
+
+            qubits = tuple(gate["qubits"])
+            gate_props = {}
+            for param in gate["parameters"]:
+                if param["name"] == "gate_error":
+                    gate_props["error"] = param["value"]
+                if param["name"] == "gate_length":
+                    gate_props["duration"] = apply_prefix(param["value"], param["unit"])
+            gates[name][qubits] = InstructionProperties(**gate_props)
+        for gate, props in gates.items():
+            if gate in name_mapping:
+                inst = name_mapping.get(gate)
+            else:
+                inst = custom_gates[gate]
+            target.add_instruction(inst, props)
+        # Create measurement instructions:
+        measure_props = {}
+        count = 0
+        for qubit in props_dict["qubits"]:
+            qubit_prop = {}
+            for prop in qubit:
+                if prop["name"] == "readout_length":
+                    qubit_prop["duration"] = apply_prefix(prop["value"], prop["unit"])
+                if prop["name"] == "readout_error":
+                    qubit_prop["error"] = prop["value"]
+            measure_props[(count,)] = InstructionProperties(**qubit_prop)
+            count += 1
+        target.add_instruction(Measure(), measure_props)
+    # Parse from configuration because properties doesn't exist
+    else:
+        for gate in conf_dict["gates"]:
+            name = gate["name"]
+            gate_props = {tuple(x): None for x in gate["coupling_map"]}
+            if name in name_mapping:
+                target.add_instruction(name_mapping[name], gate_props)
+            else:
+                custom_gate = Gate(name, len(gate["coupling_map"][0]), [])
+                target.add_instruction(custom_gate, gate_props)
+        measure_props = {(n,): None for n in range(conf_dict["n_qubits"])}
+        target.add_instruction(Measure(), measure_props)
+    # parse global configuration properties
+    dt = conf_dict.get("dt")
+    if dt:
+        target.dt = dt
+    if "timing_constraints" in conf_dict:
+        target.granularity = conf_dict["timing_constraints"].get("granularity")
+        target.min_length = conf_dict["timing_constraints"].get("min_length")
+        target.pulse_alignment = conf_dict["timing_constraints"].get("pulse_alignment")
+        target.aquire_alignment = conf_dict["timing_constraints"].get("acquire_alignment")
+    # If pulse defaults exists use that as the source of truth
+    if defs_dict is not None:
+        pulse_defs = PulseDefaults.from_dict(defs_dict)
```

Comment:
I realize this was probably in my original draft for doing this, I'm just thinking longer term it might be good to do this without using a `PulseDefaults` object. At some point we're going to have to do this when we remove this class from terra. But for this PR probably just putting a `TODO` comment is enough